sql/system_variables.go

```diff
@@ -0,0 +1,2459 @@
+		SetVarHintApplies: false,
+		Type: NewSystemBoolType("temptable_use_mmap"),
+		Default: int8(1),
+	},
+	"thread_cache_size": {
+		Name: "thread_cache_size",
+		Scope: SystemVariableScope_Global,
+		Dynamic: true,
+		SetVarHintApplies: false,
+		Type: NewSystemIntType("thread_cache_size", 0, 16384, true),
+		Default: int64(-1),
+	},
+	"thread_handling": {
+		Name: "thread_handling",
+		Scope: SystemVariableScope_Global,
+		Dynamic: false,
+		SetVarHintApplies: false,
+		Type: NewSystemEnumType("thread_handling", "no-threads", "one-thread-per-connection", "loaded-dynamically"),
+		Default: "one-thread-per-connection",
+	},
+	"thread_pool_algorithm": {
+		Name: "thread_pool_algorithm",
+		Scope: SystemVariableScope_Global,
+		Dynamic: false,
+		SetVarHintApplies: false,
+		Type: NewSystemIntType("thread_pool_algorithm", 0, 1, false),
+		Default: int64(0),
+	},
+	"thread_pool_high_priority_connection": {
+		Name: "thread_pool_high_priority_connection",
+		Scope: SystemVariableScope_Both,
+		Dynamic: true,
+		SetVarHintApplies: false,
+		Type: NewSystemIntType("thread_pool_high_priority_connection", 0, 1, false),
+		Default: int64(0),
+	},
+	"thread_pool_max_active_query_threads": {
+		Name: "thread_pool_max_active_query_threads",
+		Scope: SystemVariableScope_Global,
+		Dynamic: true,
+		SetVarHintApplies: false,
+		Type: NewSystemIntType("thread_pool_max_active_query_threads", 0, 512, false),
+		Default: int64(0),
+	},
+	"thread_pool_max_unused_threads": {
+		Name: "thread_pool_max_unused_threads",
+		Scope: SystemVariableScope_Global,
+		Dynamic: true,
+		SetVarHintApplies: false,
+		Type: NewSystemIntType("thread_pool_max_unused_threads", 0, 4096, false),
+		Default: int64(0),
+	},
+	"thread_pool_prio_kickup_timer": {
+		Name: "thread_pool_prio_kickup_timer",
+		Scope: SystemVariableScope_Global,
+		Dynamic: true,
+		SetVarHintApplies: false,
+		Type: NewSystemIntType("thread_pool_prio_kickup_timer", 0, 4294967294, false),
+		Default: int64(1000),
+	},
+	"thread_pool_size": {
+		Name: "thread_pool_size",
+		Scope: SystemVariableScope_Global,
+		Dynamic: false,
+		SetVarHintApplies: false,
+		Type: NewSystemIntType("thread_pool_size", 1, 512, false),
+		Default: int64(16),
+	},
+	"thread_pool_stall_limit": {
+		Name: "thread_pool_stall_limit",
+		Scope: SystemVariableScope_Global,
+		Dynamic: true,
+		SetVarHintApplies: false,
+		Type: NewSystemIntType("thread_pool_stall_limit", 4, 600, false),
+		Default: int64(6),
+	},
+	//TODO: implement block sizes
+	//"thread_stack": {
+	//	Name: "thread_stack",
+	//	Scope: SystemVariableScope_Global,
+	//	Dynamic: false,
+	//	SetVarHintApplies: false,
+	//	Type: NewSystemUintType("thread_stack", 131072, 18446744073709551615),
+	//	Default: uint64(286720),
+	//},
+	"time_zone": {
+		Name: "time_zone",
+		Scope: SystemVariableScope_Both,
+		Dynamic: true,
+		SetVarHintApplies: true,
+		Type: NewSystemStringType("time_zone"),
+		Default: "SYSTEM",
+	},
+	//TODO: this needs to be a function
+	"timestamp": {
```

Comment:
We have a timestamp function already

Reply:
I'll fix the comment on this, I don't mean to change this to a function, but rather that `SELECT @@timestamp` should internally call a function as the value isn't static.